### PR TITLE
properly handle initialization failures

### LINKED ISSUE #4850: Task initialization failure sets incorrect task state
## Description

If Golem task initialization fails (for example initialize function throws exception), task state will be set to _creating_ for always.

**Golem Version**:
current develop

**OS** [e.g. Windows 10 Pro]:
Ubuntu (but it shouldn't matter)

**Branch** (if launched from source):
develop

**Description of the issue**:

Golem catches exception properly, but it is handled by decorator which seems to be outdated. 
Problem occures in _task_creation_failed_ taskmanager.py line 240, where task_id parameter is set to None. This function got None, because __create_task_error_ decorator, which handles exceptions, tried to get task_id from task_dict. But task_dict doesn't have id, only task_definition does.

## Steps To Reproduce
1. Add throw statement to initialize function of any task (I tested it on transcoding use case, but should work with blender task either)
2. Request task.
3. Observe results: Golem will create task, assign id to it, but than you will see in logs: Cannot create task. task_id=None.
4. If you break somewhere with debugger you will see, that task status remained _creating_

## Expected behavior
I expect that task result will be set to _errorCreating_.

## Logs and any additional context
    INFO     [golem.task.rpc                     ] Creating task. task_dict={'type': 'FFMPEG', 'compute_on': 'cpu', 'name': 'Transcoding id 37', 'timeout': '00:30:00', 'subtask_timeout': '00:10:00', 'subtasks_count': 1, 'bid': 1, 'resources': ['/home/nieznanysprawiciel/Repos/Golem/transcoding-demo/Storage/r_25/tmp2_sptvs5'], 'options': {'output_path': '/home/nieznanysprawiciel/Repos/Golem/transcoding-demo/Storage/r_25/37', 'output_filename': 'Golem-timelapse-03-mpeg4-fullHD_1920x1080_vp9.webm', 'container': 'webm', 'video': {'resolution': [1920, 1080], 'codec': 'vp9'}}}
    INFO     [golem.task.taskmanager             ] Creating task. type=<class 'apps.transcoding.ffmpeg.task.ffmpegTask'>, id=043fed28-ff16-11e9-868b-dc2e10f8d3e6
    INFO     [golem.environments.environmentsmanager] Adding environment FFMPEG supported=<SupportStatus ok ({})>
    INFO     [golem.network.p2p.peersession      ] Starting peer session. address=94.23.196.166:40102
    INFO     [apps.transcoding.ffmpeg.utils      ] Stream /home/nieznanysprawiciel/Repos/Golem/transcoding-demo/Storage/r_25/tmp2_sptvs5 has successfully passed the extract+split operation. Segments: ['/home/nieznanysprawiciel/Data/GolemData/Requestor1/rinkeby/ComputerRes/043fed28-ff16-11e9-868b-dc2e10f8d3e6/tmp/output/tmp2_sptvs5[video-only]_0']
    ERROR    [golem.task.taskmanager             ] Cannot create task. task_id=None : Debug throw

### PR DESCRIPTION
**Pull request by etam**
fixes #4850

### CODE REVIEW

File: golem/task/taskmanager.py [hunk 1]
```diff
@@ -195,6 +195,7 @@ def create_task(self, dictionary, test=False):
             logger.info("Creating task. type=%r, id=%s", type(task), task_id)
             self.tasks[task_id] = task
             self.tasks_states[task_id] = TaskState(task)
+            dictionary['id'] = task_id
```
Comment (mfranciszkiewicz) [inline golem/task/taskmanager.py]: Shouldn't `task_id` be a keyword argument to `_create_task_error`? It isn't known before we manually add the `_create_task_error` errback. And we won't mutate input data.

File: tests/golem/task/test_rpc.py [hunk 2]
```diff
@@ -112,11 +112,14 @@ def add_new_task(task, *_args, **_kwargs):
...
-        header=mock.MagicMock(task_id='task_id'),
-    ),
-)
+    'golem.task.taskmanager.TaskManager.create_task_definition',
+    side_effect=lambda *_: (
+        Mock(task_id="task_id"),
+        Mock(side_effect=lambda *_: Mock(
+            build=Mock(side_effect=lambda: Mock(
+                header=Mock(task_id="task_id")))
+        ))
```
Comment (Wiezzel) [inline tests/golem/task/test_rpc.py]: This has definitely grown too much to stay in the decorator arguments. Please move it to the function body.

Comment (etam) [reply tests/golem/task/test_rpc.py]: To every one of them? This is on class, not function.

File: golem/task/taskmanager.py (hunk 1 above)
Comment (etam) [reply golem/task/taskmanager.py]: `_create_task_error` is used as a handler for `ClientProvider.create_task`, but `task_id` is created inside.

Comment (mfranciszkiewicz) [reply golem/task/taskmanager.py]: This doesn't solve the issue with Task API tasks anyway and it's enough to add a keyword argument in `_create_legacy_task`

File: tests/golem/task/test_rpc.py (hunk 2 above)
Comment (shadeofblue) [reply tests/golem/task/test_rpc.py]: @etam maybe the point was to move that to a separate function that you'll pass to the top `side_effect` ? ... just guessing...

Comment (Wiezzel) [reply tests/golem/task/test_rpc.py]: Sorry, I didn't express myself clearly. I was thinking about a helper function, or alternatively using `setUp` and `cleanUp` to do this.

File: golem/task/rpc.py [hunk 3]
```diff
@@ -517,28 +517,18 @@ def _create_legacy_task(
...
         task_id = task.header.task_id
 
-        try:
-            self._validate_enough_funds_to_pay_for_task(
-                task.subtask_price,
-                task.get_total_tasks(),
-                task.header.concent_enabled,
-                force
-            )
-        except Exception as exc:  # pylint: disable=broad-except
```
Comment (etam) [inline golem/task/rpc.py:527]: I deleted this, because at this point the task is not registered in TaskManager yet.

**Review by Wiezzel — APPROVED** [review]

**Review by mfranciszkiewicz — APPROVED** [review]

**Comment by etam** [comment]
I squashed commits and rebased onto latest b0.22

**Pull request merged**